rocketpy/plots/monte_carlo_plots.py

```diff
@@ -14,10 +21,185 @@ class _MonteCarloPlots:
+                    dist = (x**2 + y**2) ** 0.5
+                    # Calculate bearing: 0 is North (Y), 90 is East (X)
+                    bearing = np.degrees(np.arctan2(x, y))
+                    lat, lon = inverted_haversine(
+                        origin_lat, origin_lon, dist, bearing, earth_radius
+                    )
+                    req_lats.append(lat)
+                    req_lons.append(lon)
+
+                west, south, east, north = (
+                    min(req_lons),
+                    min(req_lats),
+                    max(req_lons),
+                    max(req_lats),
+                )
+
+                bg, mercator_extent = contextily.bounds2img(
+                    west, south, east, north, source=source_provider, ll=True
+                )
+
+                # Helper: Web Mercator (3857) to WGS84 (4326) without pyproj dependency
+                def mercator_to_wgs84(x, y):
+                    r_major = 6378137.0
+                    lon = x / r_major * 180.0 / math.pi
+                    lat = (
+                        (2 * math.atan(math.exp(y / r_major)) - math.pi / 2.0)
+                        * 180.0
+                        / math.pi
+                    )
+                    return lat, lon
+
+                # Convert corners of the fetched image
+                bg_lat_min, bg_lon_min = mercator_to_wgs84(
+                    mercator_extent[0], mercator_extent[2]
+                )  # Bottom-Left
+                bg_lat_max, bg_lon_max = mercator_to_wgs84(
+                    mercator_extent[1], mercator_extent[3]
+                )  # Top-Right
+
+                # Calculate X/Y meters relative to origin (lat0, lon0) using haversine
+                # X = Distance along longitude (East-West)
+                # Y = Distance along latitude (North-South)
+
+                # Calculate X min (Left)
+                x_min = haversine(
+                    origin_lat, origin_lon, origin_lat, bg_lon_min, earth_radius
+                )
+                if bg_lon_min < origin_lon:
+                    x_min = -x_min
+
+                # Calculate X max (Right)
+                x_max = haversine(
+                    origin_lat, origin_lon, origin_lat, bg_lon_max, earth_radius
+                )
+                if bg_lon_max < origin_lon:
+                    x_max = -x_max
+
+                # Calculate Y min (Bottom)
+                y_min = haversine(
+                    origin_lat, origin_lon, bg_lat_min, origin_lon, earth_radius
+                )
+                if bg_lat_min < origin_lat:
+                    y_min = -y_min
+
+                # Calculate Y max (Top)
+                y_max = haversine(
+                    origin_lat, origin_lon, bg_lat_max, origin_lon, earth_radius
+                )
+                if bg_lat_max < origin_lat:
+                    y_max = -y_max
+
+                return bg, [x_min, x_max, y_min, y_max]
+
+            except Exception as e:
```

Comment:
I wonder if this exception is really necessary... If you simply remove it, the code will break and user will try to fix their inputs or run the code again without the background option. It's better to not create "silent errors" within the code. So my suggestion is for us to remove the exception